### DISCUSSION TITLE
Is there an easy way to completely ignore namespaces?

### DISCUSSION THREAD

**Discussion by rrenna** [pdvrieze/xmlutil#162] [Q&A]
I have value objects which are reused in three different types of XML documents I'm parsing.

One of the three types needs to be written to, but the other are read-only. All three have different `xmlns` values - which is causing parsing issues depending on which one I set in the shared classes (in `@XmlSerialName`).

Is there a way to only set it correctly for the 3rd type of document I need to actually recreate and write, and completely ignore this for the other two cases?

**Comment by pdvrieze** [comment]
The way to do this is to use a filter for the input. You'll have to get the parser by hand (easiest using the `XmlStreaming` object, create a reader that transforms all namespaces in it to the null namespace (and null prefix), and then pass that wrapping reader to the serialization framework.
